adapters/invibes/invibes.go

```diff
@@ -0,0 +1,327 @@
+		},
+	}
+
+	for _, imp := range request.Imp {
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errors = append(errors, &errortypes.BadInput{
+				Message: "Error parsing bidderExt object",
+			})
+		}
+		if len(errors) == 0 {
+			var invibesExt openrtb_ext.ExtImpInvibes
+			if err := json.Unmarshal(bidderExt.Bidder, &invibesExt); err != nil {
+				errors = append(errors, &errortypes.BadInput{
+					Message: "Error parsing invibesExt parameters",
+				})
+			}
+			if imp.Banner == nil {
+				errors = append(errors, &errortypes.BadInput{
+					Message: "Banner not specified",
+				})
+			}
+			if len(errors) == 0 {
+				currentBanner := *imp.Banner
+				var adFormats []openrtb.Format
+				if currentBanner.Format != nil {
+					adFormats = currentBanner.Format
+				} else if currentBanner.W != nil && currentBanner.H != nil {
+					adFormats = []openrtb.Format{
+						{
+							W: *currentBanner.W,
+							H: *currentBanner.H,
+						},
+					}
+				}
+
+				invibesInternalParams.Host = invibesExt.Host
+				invibesInternalParams.BidParams.PlacementIds = append(invibesInternalParams.BidParams.PlacementIds, strings.TrimSpace(invibesExt.PlacementId))
+				invibesInternalParams.BidParams.Properties[invibesExt.PlacementId] = InvibesPlacementProperty{
+					ImpId:   imp.ID,
+					Formats: adFormats,
+				}
+				if invibesExt.Debug.TestIp != "" {
+					invibesInternalParams.TestIp = invibesExt.Debug.TestIp
+				}
+				if invibesExt.Debug.TestBvid != "" {
+					invibesInternalParams.TestBvid = invibesExt.Debug.TestBvid
+				}
+				invibesInternalParams.TestLog = invibesExt.Debug.TestLog
+			}
+		}
+	}
+	if reqInfo.PbsEntryPoint == pbsmetrics.ReqTypeAMP {
+		invibesInternalParams.IsAmp = true
+	}
+
+	if len(invibesInternalParams.BidParams.PlacementIds) == 0 {
+		return nil, errors
+	}
+
+	invibesInternalParams.Gdpr = gdprApplies
+	invibesInternalParams.GdprConsent = consentString
+
+	newHttpRequest, err := a.makeRequest(invibesInternalParams, reqInfo, httpRequests, request)
+	if err != nil {
+		errors = append(errors, err)
+	} else if newHttpRequest != nil {
+		httpRequests = append(httpRequests, newHttpRequest)
+	}
+
+	return httpRequests, errors
+}
+
+func (a *InvibesAdapter) makeRequest(invibesParams InvibesInternalParams, reqInfo *adapters.ExtraRequestInfo, existingRequests []*adapters.RequestData, request *openrtb.BidRequest) (*adapters.RequestData, error) {
+
+	url, err := a.makeURL(request, invibesParams.Host)
+	if err != nil {
+		return nil, err
+	}
+	parameter, errp := a.makeParameter(invibesParams, request)
+	if errp != nil {
+		return nil, errp
+	}
+	body, errm := json.Marshal(parameter)
+	if errm != nil {
+		return nil, errm
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	if request.Device != nil {
+		headers.Add("User-Agent", request.Device.UA)
+	}
+
+	if invibesParams.TestIp != "" {
```

Comment:
Is the test ip functionality necessary in the production adapter?

Reply:
No is not necessary. We have removed it